### PR TITLE
ov2.0 text detection demo

### PR DESCRIPTION
**Pull request by vladimir-dudnik**
migrate to ov2.0 api
fixed docs
style

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
need 2022.1.582 to buid

**Comment by vladimir-dudnik** [comment]
unrelated issue with background model

20:47:42  [ INFO ] OpenVINO Inference Engine
20:47:42  [ INFO ] 	build: 2022.1.0-6610-cc602ac6fda
20:47:42  [ INFO ] Reading model /tmp/tmpvfhlyfe9/models/intel/instance-segmentation-person-0007/FP16/instance-segmentation-person-0007.xml
20:47:42  [ WARNING ] The "--background" option works only for ImageMattingWithBackground model. Option will be omitted.
20:47:42  [ INFO ] 	Input layer: image, shape: [1, 3, 320, 544], precision: f32, layout: NCHW
20:47:42  [ INFO ] 	Output layer: boxes, shape: [0, 5], precision: f32, layout: 
20:47:42  [ INFO ] 	Output layer: 1973, shape: [0], precision: i64, layout: 
20:47:42  [ INFO ] 	Output layer: masks, shape: [0, 224, 224], precision: f32, layout:

**Comment by Wovchena** [comment]
Jenkins please retry a build

File: demos/text_detection_demo/cpp/include/text_recognition.hpp [hunk 1]
```diff
@@ -7,6 +7,47 @@
...
+
+#include "cnn.hpp"
+
+std::string CTCGreedyDecoder(const std::vector<float>& data, const std::string& alphabet, char pad_symbol, double* conf);
+std::string CTCBeamSearchDecoder(const std::vector<float>& data, const std::string& alphabet, char pad_symbol, double* conf, int bandwidth);
+std::string SimpleDecoder(const std::vector<float>& data, const std::string& alphabet, char pad_symbol, double* conf, int start_index);
+
+
+class DecoderNotFound {};
+
```
Comment (Wovchena) [inline demos/text_detection_demo/cpp/include/text_recognition.hpp:20]:
```suggestion
```

File: models/intel/text-recognition-0014/README.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ The net output is a blob with name `logits` and the shape `16, 1, 37` in the for
 
 - `W` - output sequence length
 - `B` - batch size
-- `L` - confidence distribution across alphanumeric symbols: `#0123456789abcdefghijklmnopqrstuvwxyz`, where # - special blank character for CTC decoding algorithm.
+- `L` - confidence distribution across alphanumeric symbols: `0123456789abcdefghijklmnopqrstuvwxyz`, where # - special blank character for CTC decoding algorithm.
```
Comment (Wovchena) [inline models/intel/text-recognition-0014/README.md:43]: Without `#` in the list, the part
> where # - special blank character for CTC decoding algorithm

looks strange

Comment (vladimir-dudnik) [reply models/intel/text-recognition-0014/README.md]: ok, changed

**Review by Wovchena — APPROVED** [review]

**Review by morkovka1337 — APPROVED** [review]
